Documentation/Browser/BrowsingContextsAndNavigables.md

```diff
@@ -0,0 +1,128 @@
+additional Environment Records required to access names in the current scope, any running generator state,
+and most importantly to the thread analogy, the state needed to suspend and resume execution of that script.
+Throughout the ECMAScript specification, the text refers to a stack of execution contexts, with the uppermost
+one always being the [*running execution context*](https://tc39.es/ecma262/#running-execution-context).
+As with Environment Records, Realms and Execution contexts are not visible to running JavaScript code.
+
+The final missing piece for the JavaScript execution model is how these stacks of Execution Contexts
+are actually scheduled to run by the ECMAScript implementation. In the most common case, this means directly
+mapping the ECMAScript model to the earlier native concepts of threads and processes in a way that
+allows for flexibility in the implementation strategies. The last thing that the specification authors want
+to do is constrain implementations so much that innovation and experimentation becomes impossible.
+The method for this mapping is the two related specification objects [*Agents*](https://tc39.es/ecma262/#sec-agents)
+and [*Agent Clusters*](https://tc39.es/ecma262/#sec-agent-clusters). The Execution Context stack mentioned
+above actually belongs to an Agent, which holds said stack, a set of metadata about the memory model,
+and a shared reference to an [*executing thread*](https://tc39.es/ecma262/#executing-thread). To re-iterate,
+an Agent holds a stack of Execution Contexts, with the topmost entry being the running execution context.
+Each Execution Context holds a Realm and a specific script's context, including the current function and
+any state required to pause and resume the execution for that context. The Realm holds the Global
+Object for the Execution Context, and any ECMAScript or host-specific intrinsics required to create the
+desired environment for code to run in. More loosely, an Agent is a specification artefact that somewhat
+maps the execution of a JavaScript script or module to a native thread of execution. But the specification
+does so in a way that allows a host/embedder to choose to switch out which Agent is currently executing
+its running execution context on that native thread.
+
+SharedArrayBuffers and Atomics add a special kind of wrinkle to the ECMAScript specification. Defining
+how that work required the formalization of a memory model, similar to what C++11 and C11 and Java 5 had
+to do before them. The Agent Cluster is the formalism that ties the memory model back to the execution
+model. As described in the specification, an Agent Cluster is a set of Agents that can communicate
+via shared memory. The exact mechanism is unspecified, but the hard rule is that all Agents within
+a particular Agent Cluster must observe the same order of reads and writes to SharedArrayBuffers and
+as a result of ECMAScript Atomic objects.
```

Comment:
Sounds correct, but my experience with shared memory in JS implementations is ~0 :)